src/base/bittorrent/infohash.cpp

```diff
@@ -28,68 +28,14 @@
 
 #include "infohash.h"
 
-#include <QByteArray>
-#include <QHash>
+const int InfoHashTypeId = qRegisterMetaType<BitTorrent::InfoHash>();
 
-using namespace BitTorrent;
-
-const int InfoHashTypeId = qRegisterMetaType<InfoHash>();
-
-InfoHash::InfoHash(const lt::sha1_hash &nativeHash)
-    : m_valid(true)
-    , m_nativeHash(nativeHash)
-{
-    const QByteArray raw = QByteArray::fromRawData(nativeHash.data(), length());
-    m_hashString = QString::fromLatin1(raw.toHex());
-}
-
-InfoHash::InfoHash(const QString &hashString)
-    : m_valid(false)
-{
-    if (hashString.size() != (length() * 2))
-        return;
-
-    const QByteArray raw = QByteArray::fromHex(hashString.toLatin1());
-    if (raw.size() != length())  // QByteArray::fromHex() will skip over invalid characters
-        return;
-
-    m_valid = true;
-    m_hashString = hashString;
-    m_nativeHash.assign(raw.constData());
-}
-
-bool InfoHash::isValid() const
-{
-    return m_valid;
-}
-
-InfoHash::operator lt::sha1_hash() const
-{
-    return m_nativeHash;
-}
-
-InfoHash::operator QString() const
-{
-    return m_hashString;
-}
-
-bool BitTorrent::operator==(const InfoHash &left, const InfoHash &right)
-{
-    return (static_cast<lt::sha1_hash>(left)
-            == static_cast<lt::sha1_hash>(right));
-}
-
-bool BitTorrent::operator!=(const InfoHash &left, const InfoHash &right)
-{
-    return !(left == right);
-}
-
-bool BitTorrent::operator<(const InfoHash &left, const InfoHash &right)
+BitTorrent::InfoHash BitTorrent::InfoHash::fromString(const QString &hashString)
 {
-    return static_cast<lt::sha1_hash>(left) < static_cast<lt::sha1_hash>(right);
+    return BitTorrent::InfoHash {SHA1Hash::fromString(hashString)};
```

Comment:
is it possible to write `return {SHA1Hash::fromString(hashString)};`?

Reply:
>is it possible to write `return {SHA1Hash::fromString(hashString)};`?

Seems Yes.